src/main/scala/stage/phases/PreElaboration.scala

```diff
@@ -20,10 +26,11 @@ class PreElaboration extends Phase with PreservesAll[Phase] with HasRocketChipSt
 
   override def transform(annotations: AnnotationSeq): AnnotationSeq = {
 
+    val stageOpts = view[StageOptions](annotations)
     val rOpts = view[RocketChipOptions](annotations)
     val topMod = rOpts.topModule.get
 
-    val config = getConfig(rOpts.configNames.get)
+    val config = getConfig(rOpts.configNames.get) ++ WithTargetDir(stageOpts.targetDir)
```

Comment:
```
val config = getConfig(rOpts.configNames.get).alterPartial {
  case TargetDirKey => stageOpts.targetDir
}
```

Reply:
I agree that this is much better.  I'll make the change.